### DISCUSSION TITLE
Aside on PR #10184

### DISCUSSION THREAD

**Discussion by RandomDSdevel** [Homebrew/discussions#400] [Tap maintenance and Homebrew development] [answered]
_Originally posted by @Rylan12 in https://github.com/Homebrew/brew/issues/10184#issuecomment-752745252_:  

> CI (on `ubuntu-20.04`) is now failing with the following message:
>
> ```
> In /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils/ruby.sh line 20:
>     for ruby_exec in $(which -a ruby 2>/dev/null) $(PATH=$HOMEBREW_PATH which -a ruby 2>/dev/null)
>                        ^---^ SC2230: which is non-standard. Use builtin 'command -v' instead.
>                                                                         ^---^ SC2230: which is non-standard. Use builtin 'command -v' instead.
>
> For more information:
>   https://www.shellcheck.net/wiki/SC2230 -- which is non-standard. Use builti...
> Error: which is non-standard. Use builtin 'command -v' instead.
> ```
>
> According to https://www.shellcheck.net/wiki/SC2230, this is opt-in only and I don't believe we've opted in... Odd that this is only the case on Ubuntu 20.04.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;I just wanted to note out-of-band that I've also seen this warning when running '`brew tests`' while preparing [recent PRs](https://github.com/search?q=org%3AHomebrew+is%3Apr+author%3ARandomDSdevel+merged%3A%3E2020-12-01+&type=) of my own for other things.  Granted, the setup I've been limping along on is _aaaaaaan_-cient, at least in 'computer years,' and thus more than a bit of an outlier.  (I'm still on OS X v10.11.6 'El Capitan,' build 15G22010, so feel free to ignore me; I just wanted to add another data point.)

**Comment by RandomDSdevel** [comment]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;For reference, my full '`brew config`' as of right now (I haven't run '`brew update` yet today) is:  

```
…~ …$ brew config
HOMEBREW_VERSION: 2.7.1
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: fdb71a5cb3029da2ef74d7e09c40575bbfb4ec72
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 963d33f8193d0a1537a8dada41fa7938085325ec
Core tap last commit: 15 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: ["--appdir=/Applications", "--prefpanedir=/Library/PreferencePanes", "--colorpickerdir=/Library/ColorPickers", "--dictionarydir=/Library/Dictionaries", "--servicedir=/Library/Services", "--qlplugindir=/Library/QuickLook", "--fontdir=/Library/Fonts", "--input_methoddir=/Library/Input Methods", "--screen_saverdir=/Library/Screen Savers", "--internet_plugindir=/Library/Internet Plug-Ins"]
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.lcUd9snI2W/org.macosforge.xquartz:0
HOMEBREW_EDITOR: nano --fill=100
HOMEBREW_FORCE_BREWED_CURL: set
HOMEBREW_FORCE_BREWED_GIT: set
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 2
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_UPDATE_TO_TAG: set
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: dual-core 64-bit core2
Clang: 8.0 build 800
Git: 2.20.1 => /usr/local/opt/git/bin/git
Curl: 7.63.0 => /usr/local/opt/curl/bin/curl
Java: 11.0.2
macOS: 10.11.6-x86_64
CLT: 8.2.0.0.1.1480973914
Xcode: 8.2.1
XQuartz: 2.7.11 => /opt/X11
```

(Yes, I'm aware my Java installation is out of date.  That doesn't matter in this context, though.)

**Comment by Rylan12** [comment, marked as answer]
Okay, I finally got around to troubleshooting this. Turns out the issue is that the version of `shellcheck` was too old.

You need to use a version of `shellcheck` that's either `< 0.5.0` or `>= 0.7.1`. My recommendation is to use the most recent version of `shellcheck` (which happens to be 0.7.1).

For more info see https://github.com/Homebrew/brew/pull/10192#issuecomment-753285580

So, to solve your problem, you should just need to update `shellcheck` to the latest version.

Interestingly, though, I didn't see this issue when running `brew tests`...
